src/script/interpreter.cpp

```diff
@@ -2385,6 +2385,24 @@ bool GenericTransactionSignatureChecker<T>::CheckSequence(const CScriptNum& nSeq
     return true;
 }
 
+template <class T>
+const CTransaction* GenericTransactionSignatureChecker<T>::GetTransaction() const
+{
+    return (CTransaction*)(txTo);
```

Comment:
Can you use `static_cast<CTransaction*>` rather than a C-style cast? This will get some compile-time checking that the cast makes sense.

You can also use the CTransaction constructor, like `return CTransaction(*txTo);` although I expect this will make a copy.

Reply:
Using static_cast

Reply:
Switched back to original implementation that copies the tx weight calculation from `src/consensus/validation.cpp`.

Reply:
Cool. Makes sense, given offline discussion about what our options are here.